Ghidra/Features/Base/src/main/java/ghidra/program/flatapi/FlatProgramAPI.java

```diff
@@ -1514,6 +1514,35 @@ public final List<Symbol> getSymbols(String name, Namespace namespace) {
 	public final Namespace getNamespace(Namespace parent, String namespaceName) {
 		return currentProgram.getSymbolTable().getNamespace(namespaceName, parent);
 	}
+	
+	/**
+	 * Creates a new namespace with the given name contained inside the
+	 * specified parent namespace.
+	 * Pass <code>null</code> for namespace to indicate the global namespace.
+	 * If the namespace already exists then the existing one will be returned.
+	 * @param parent the parent namespace, or null for global namespace
+	 * @param namespaceName the requested namespace's name
+	 * @return the namespace with the given name
+	 * @throws InvalidInputException if the name is invalid
+	 */
+	public final Namespace createNamespace(Namespace parent, String namespaceName)
+			throws InvalidInputException {
+		SymbolTable symbolTable = currentProgram.getSymbolTable();
+		Namespace ns = symbolTable.getNamespace(namespaceName, parent);
+		if (ns != null) {
+			SymbolType type = ns.getSymbol().getSymbolType();
+			if (type == SymbolType.NAMESPACE || type == SymbolType.CLASS) {
+				return ns;
+			}
+		}
+		try {
+			return symbolTable.createNameSpace(parent, namespaceName, SourceType.USER_DEFINED);
+		}
+		catch (DuplicateNameException e) {
+			// impossible
+			throw new AssertException(e);
```

Comment:
Throwing this assert is dependent on the naming rules of other symbol types which could conflict (e.g., Library).  Method should probably throw DuplicateNameException

Reply:
It would have been obtained from the call to `SymbolTable::getNamespace` above and returned. Any existing symbol type which would conflict *should* have an underlying namespace which would make this condition unreachable.

Reply:
I ran a sanity check and you were correct. I think I had the behavior of mixed and thought you could have a namespace with the same name as a library like you can with a function and label.